library/common/http/dispatcher.cc

```diff
@@ -251,31 +343,18 @@ envoy_status_t Dispatcher::resetStream(envoy_stream_t stream) {
   post([this, stream]() -> void {
```

Comment:
We need to update stream state (`closed`) and fire callbacks synchronously here, prior to dispatching the reset.

Reply:
This is one of the biggest things I want to discuss. Because (related to above) if we keep all the cancellation state in code executed in Envoy's main thread, couldn't we relax the atomic boolean to just a simple boolean?

Reply:
from offline: the contract with cancellation is that it is an interrupt. In other words, by the time this function returns no further callbacks associated with this stream will be surfaced. Hence we can't dispatch exchanging the closed state. It has async logic in it but the effect is sync.

Reply:
@goaway This is the only point that remains from our conversation yesterday. The main wrinkle in moving the dispatchable call outside of the posted body is that up until now we asserted that all DirectStream interaction happened in the threading context of the event dispatcher. 

By moving the dispatchable call outside of the post body we are now interacting with the DirectStream in the platform threading context.